### DISCUSSION TITLE
Add the  possibility to use custom server jars

### DISCUSSION THREAD

**Discussion by TropicLegend** [jeremylvln/Shulker#252] [Ideas]
It would be very beneficial to have the ability to use custom server jars since only paper or folia feels limiting

**Comment by jeremylvln** [comment]
Hi!

Shulker uses `docker-minecraft-server` by itzg under the hood which supports a lot of platforms: https://docker-minecraft-server.readthedocs.io/en/latest/types-and-platforms/. However, each platform needs a Shulker companion/agent plugin to hook it to the cluster.

Currently, I've written one only for Bukkit/Spigot/Paper/Folia, that's why there are the only ones supported currently. Bukkit and Spigot are not supported as-it because Paper cover them without hassle.

Someone might want to have modded servers (Forge/Fabric whatever), but for now I want to focus myself on hardening Shulker and its current features rather than looking to introduce new ones

**Comment by TropicLegend** [comment]
I specifically asked because I would like to use a paper fork and I am very confident it would still work with said Shulker companion/agent.
I am talking about purpur and aspaper and aspurpur.

Especially aspaper and aspurpur could be very interesting because they could solve our problem related to maps and persistence.

Source: https://github.com/InfernalSuite/AdvancedSlimePaper

Downloads can only be found in their discord I think.

**Reply by jeremylvln** [reply]
I have the solution in my head, I can manage to add that :)
Something like:

```yaml
version:
  channel: Paper
  customJar: # a ResourceRef like world/plugins/patches
    url: https://blabla
```

I will still need the channel to know what agent to inject into the server.

**Comment by TropicLegend** [comment]
They use a different world region format called slime regions and they are stored in databases. It is really interesting technology

**Reply by jeremylvln** [reply]
Wait this is actually insane o m g
I'm really interested myself in this tech', so you have more documentation links about that (like an official spec of the file format, etc..)? That's crazy

EDIT: found https://hypixel.net/threads/dev-blog-5-storing-your-skyblock-island.2190753/

**Reply by TropicLegend** [reply]
https://github.com/InfernalSuite/
This is the current repository :)

**Comment by TropicLegend** [comment]
On the other hand, I can totally see how there are more important things and features.
Let's not rush things :+1:

**Comment by jeremylvln** [comment]
Closing as it is accepted as an issue #253
